src/openeo_gfmap/manager/job_manager.py

```diff
@@ -439,6 +551,12 @@ def _write_stac(self):
         self._root_collection.normalize_hrefs(str(root_path))
         self._root_collection.save(catalog_type=CatalogType.SELF_CONTAINED)
 
+    def _persist_stac(self):
+        """Persists the STAC collection by saving it into a binary file."""
+        _log.info("Persisting STAC collection to temp file %s.", self._catalogue_cache)
+        with open(self._catalogue_cache, "wb") as file:
```

Comment:
Perhaps good to add some resilience here by performing a check on self_catalogue_cache